app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/SaveInStorePickupPlugin.php

```diff
@@ -30,14 +44,26 @@ public function beforeSave(
         SourceRepositoryInterface $subject,
         SourceInterface $source
     ): array {
+
+        if (!$source instanceof DataObject) {
+            return [$source];
+        }
+
         $extensionAttributes = $source->getExtensionAttributes();
+        if ($extensionAttributes === null) {
+            $extensionAttributes = $this->extensionAttributesFactory->create(SourceInterface::class);
+            /** @noinspection PhpParamsInspection */
+            $source->setExtensionAttributes($extensionAttributes);
+        }
 
-        if ($extensionAttributes !== null && $source instanceof  DataObject) {
-            $source->setData(Location::IS_PICKUP_LOCATION_ACTIVE, $extensionAttributes->getIsPickupLocationActive());
-            $source->setData(Location::FRONTEND_NAME, $extensionAttributes->getFrontendName());
-            $source->setData(Location::FRONTEND_DESCRIPTION, $extensionAttributes->getFrontendDescription());
+        if (empty($extensionAttributes->getFrontendName())) {
+            $extensionAttributes->setFrontendName($source->getName());
         }
 
+        $source->setData(Location::IS_PICKUP_LOCATION_ACTIVE, $extensionAttributes->getIsPickupLocationActive());
```

Comment:
May we proceed with these steps only if Extension Attributes were passed?
Except the Frontend Name.

Extension Attributes Passed -> use Frontend Name from extension attributes.
Extension Attributes do not Passed -> Source does not have Frontend Name -> use Source Name
Extension Attributes do not Passed -> Source  has Frontend Name -> do not change the value